### ISSUE TITLE
Sort tags support

### ISSUE THREAD

**Issue by snoozysoft** [OxygenCobalt/Auxio#172]
#### Describe the feature you want to implement:
ID3 has tags specifically for telling a music player/library where to sort the albums. This is largely useful for albums or artists that are tagged in the artists native language e.g. Japanese, since most if not all music players I have seen and used automatically put them all at the bottom of the library due to them not being Latin script. Acknowledging these tags (artistsort, albumsort, albumartistsort) would put these albums in their proper place as dictated by the user.

#### Is your feature request related to a problem? Please describe:
As mentioned above, album titles that are in a language other than one that uses Latin script get sent to the bottom of the library.

#### Do other music players handle this? If so, how?
GoneMAD Music Player supports it; it reads the sort tags and automatically puts them in the appropriate place alphabetically. I use romanized transliterations of the album titles in the sort tags so that they appear in roughly the correct spot. E.g. "新青年" by NINGEN-ISU gets placed in the library as if it were tagged "Shin Seinen".

#### Why do you think this will improve everyone's usage of Auxio?
It's just a very handy and useful little tag that not many people know about. Should be easy enough in theory to implement and will make the QoL better overall for any potential user. Also useful for people like me who are sticklers for properly tagging artists and albums in their appropriate langauges/stylizations.

#### Due Diligence:
- [*] I have read the [Contribution Guidelines](https://github.com/OxygenCobalt/Auxio/blob/dev/.github/CONTRIBUTING.md).
- [*] I have read the [Accepted Additions and Requests](https://github.com/OxygenCobalt/Auxio/blob/dev/info/ADDITIONS.md) document.
- [*] I have checked for this feature in the [FAQ](https://github.com/OxygenCobalt/Auxio/blob/dev/info/FAQ.md).

**Comment by snoozysoft** [comment]
Fingers crossed you can eventually! I'll be keeping a close eye on this project until then, cos this thing is otherwise exactly what I want out of a music player. Great work so far <3

**Comment by OxygenCobalt** [comment]
This will arrive in the next Auxio release. If you want, please feel free to try out this debug APK (Rename the extension from ZIP).

[Auxio_SortTags.zip](https://github.com/OxygenCobalt/Auxio/files/9107856/Auxio_SortTags.zip)

Please report any bugs, alongside attaching an audio file so I can reproduce it. Note that album artist sort tags will only be used in the presence of a corresponding album artist tag, and sort tags may not work if only some songs in an album have them.

**Comment by OxygenCobalt** [comment]
Quick question @snoozysoft, is it correct to take sort tags into account in search? That way, you could search for "Shin Seinen" and then get back "新青年", without having to resort to typing japenese characters.

**Comment by snoozysoft** [comment]
Ooh, that sounds a good idea. Unsure if GoneMAD does it and I can't check rn but it definitely seems like a useful addition.

Just to double check did you implement just Vorbis' sort tags or did you get both that and ID3's TSO(X) tags too?

**Comment by OxygenCobalt** [comment]
Cool, I'll try going for it then.

As for the tags, I got them both. You can look in my code [here](https://github.com/OxygenCobalt/Auxio/blob/7833ec4460a175783b6dc678d0e74b35a4684719/app/src/main/java/org/oxycblt/auxio/music/system/ExoPlayerBackend.kt#L207) and check if I got the sort tags right. I am aware that there are some other variants (FFMPEG rights `SORT-ARTIST` for vorbis) that I might add later if they are widely used.

**Comment by snoozysoft** [comment]
Awesome! It looks correct to me from what I can tell. Those should be the two most used since I believe those are generally what most taggers will use for ID3 and Vorbis; I'm not too aware of other ones myself. Will grab that test build and give it a go.

**Comment by snoozysoft** [comment]
That'd explain it, my phone only goes into universal dark mode at evening. Checking it now confirms that! Keep up the good work 💜

**Comment by OxygenCobalt** [comment]
Okay, I was able to identify one sort tag bug. With artists, if a song has an `album artist`, `artist`, and `artist sort` tag, but no `album artist sort` tag,  `artist sort` will be incorrectly chosen, even though `album artist` will be used for the actual artist entry. 

I have already fixed this, `album artist sort` will only be applied in the presence of `album artist`, and `artist sort` will only be applied in the presence of `artist`. Artists without either type of name (i.e Unknown Artist) will never use sort tags, as I don't think that makes sense.

I've also implemented sort tag (and file name) comparisons into search.
